### PR TITLE
CRN validation

### PR DESCRIPTION
**Pull request by ujjwal-ibm**

### CODE REVIEW

File: builder/ibmcloud/vpc/step_get_subnet_info.go [hunk 1]
```diff
@@ -43,6 +45,63 @@ func (s *stepGetSubnetInfo) Run(_ context.Context, state multistep.StateBag) mul
...
+
+		searchURL := "https://api.global-search-tagging.cloud.ibm.com"
+		globalSearchV2Options := &searchv2.GlobalSearchV2Options{
+			URL:           searchURL,
+			Authenticator: vpcService.Service.Options.Authenticator,
+		}
+		globalSearchAPIV2, err := searchv2.NewGlobalSearchV2(globalSearchV2Options)
+
+		if err != nil {
+			fmt.Println("Gen2 Service creation failed.", err)
```
Comment (astha-jain) [inline builder/ibmcloud/vpc/step_get_subnet_info.go:59]: Incorrect error message

File: builder/ibmcloud/vpc/step_get_subnet_info.go [hunk 2]
```diff
@@ -43,6 +45,63 @@ func (s *stepGetSubnetInfo) Run(_ context.Context, state multistep.StateBag) mul
...
+			query := fmt.Sprintf("crn:\"%s\"", crnToCheck)
+			isHidden := "any"
+			searchOptions := &searchv2.SearchOptions{
+				Query:    &query,
+				IsHidden: &isHidden,
+			}
+			res, _, _ := globalSearchAPIV2.Search(searchOptions)
+			if len(res.Items) != 0 {
+				ui.Say(fmt.Sprintf("%s Encryption information successfully retrieved ...", res.Items[0].GetProperty("name")))
+			}
```
Comment (astha-jain) [inline builder/ibmcloud/vpc/step_get_subnet_info.go:101]: What if global catalog does not contain the CRN? Need to put in error handling for if no item is found

File: builder/ibmcloud/vpc/step_verify_input.go [hunk 3]
```diff
@@ -139,6 +142,74 @@ func (s *stepVerifyInput) Run(_ context.Context, state multistep.StateBag) multi
 			state.Put("user_sec_grp_vpc", *secGrp.VPC.ID) // check for vpc is done as part of subnet fetch.
 		}
 	}
+
+	// crn validation
+
+	if config.CatalogOfferingCRN != "" || config.CatalogOfferingVersionCRN != "" || config.EncryptionKeyCRN != "" {
```
Comment (BShrivastav) [inline builder/ibmcloud/vpc/step_verify_input.go:148]: Can we provide `CatalogOfferingVersionCRN` and `CatalogOfferingCRN` together ?

File: developer/Makefile [hunk 4]
```diff
@@ -29,6 +29,10 @@ run-catalog:
 	cd ..; packer validate -var-file="developer/variables-catalog-version.pkrvars.hcl" developer/examples/build.vpc.catalogimage-version.pkr.hcl
 	cd ..; packer build -var-file="developer/variables-catalog-version.pkrvars.hcl" developer/examples/build.vpc.catalogimage-version.pkr.hcl
 	
+run-catalog-version:
```
Comment (BShrivastav) [inline developer/Makefile:32]: Could we also have a command to support `catalog-offering-crn` ?
Also what would be the use case of running - `run-catalog`?

**Review by BShrivastav — COMMENTED** [review]
Proposed couple suggestions along with the mutual exclusion check for the catalog-version and catalog-offering CRNs

File: builder/ibmcloud/vpc/step_verify_input.go (hunk 3 above)
Comment (ujjwal-ibm) [reply builder/ibmcloud/vpc/step_verify_input.go:148]: why ? the config validation should happen before input validation

**Review by astha-jain — APPROVED** [review]

**Pull request merged**